lib/components/BraidLoadableProvider/BraidLoadableProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ReactNode } from 'react';
+import loadable from 'sku/@loadable/component';
+import {
+  BraidProvider,
+  BraidProviderProps,
+} from '../BraidProvider/BraidProvider';
+
+// @ts-ignore loadable can not currently type dynamic props
+const BraidLoadable = loadable.lib((props: { themeName: string }) =>
+  import(`../../../themes/${props.themeName}`),
+);
+
+interface BraidLoadableProviderProps extends Omit<BraidProviderProps, 'theme'> {
+  themeName: string;
+  fallback?: ReactNode;
+}
+export const BraidLoadableProvider = ({
+  themeName,
+  children,
+  ...restProps
+}: BraidLoadableProviderProps) => (
+  <BraidLoadable themeName={themeName}>
+    {({ default: theme }: any) => (
+      <BraidProvider theme={theme} {...restProps}>
```

Comment:
If you put `{...restProps}` before the `theme` prop, it'll ensure that users can't override the theme.